.circleci/config.yml

```diff
@@ -0,0 +1,69 @@
+defaults: &DEFAULT
+  working_directory: /home/circleci/xblock-html
+  docker:
+    - image: circleci/python:2.7
+
+
+version: 2
+jobs:
+  build:
+    <<: *DEFAULT
+    steps:
+      - checkout
+      - restore_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "requirements/test.txt" }}
+      - run:
+          name: Installing requirements
+          command: |
+            [ ! -f /tmp/workspace/env/bin/activate ] && virtualenv /tmp/workspace/env
+            source /tmp/workspace/env/bin/activate
+            pip install -r requirements/test.txt
+      - save_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "requirements/test.txt" }}
+          paths:
+            - env
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - env
+
+  test:
+    <<: *DEFAULT
+    steps:
+      - checkout
+      - attach_workspace:
+          at: /home/circleci/xblock-html
+      - run:
+          name: Activate virtualenv
+          command: echo 'source /tmp/workspace/env/bin/activate' >> $BASH_ENV
+      - run:
+          name: Running tests
+          command: tox
+
+  quality:
+    <<: *DEFAULT
+    steps:
+      - checkout
+      - attach_workspace:
+          at: /home/circleci/xblock-html
+      - run:
+          name: Activate virtualenv
+          command: echo 'source /tmp/workspace/env/bin/activate' >> $BASH_ENV
+      - run:
+          name: Test Quality
+          command: TOXENV=quality && tox
+      - run:
+          name: Generating code coverage
+          command: codecov
```

Comment:
`codecov` only uploads results from `pytest-cov` to cloud in order to be visible to everybody (to include coverage report to `README.md` or etc).

It uses coverage 'artifacts' which were generated by `pytest-cov`. 
No config is needed for open source projects.
There is no point to run it locally because it will update coverage based on 'uncommited' code.

Reply:
I am not sure how is codecov going to get the artifacts from a parallel job (which may not even have completed at this point). Or am I missing something?

We need local diff coverage so that when doing dev one can see if the tests have added the necessary coverage to the code that has been added. Can this do that? If not, we need to add diff-cover like in the blockstore repo I linked to.

Reply:
Okay, I realize the later was not part of the scope of the task but it is related so lets get it in as well.

Reply:
Yeah, @symbolist you are right, i've moved `codecov` to other task and added local `diff_cover`.